qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1420,6 +1420,38 @@ def collect_runs(self, namelist):
                     group_list.append(tuple(group))
         return set(group_list)
 
+    # pylint: disable=too-many-boolean-expressions
+
+    def collect_1q_runs(self):
+        """Return a set of non-conditional runs of 1q "op" nodes."""
+        group_list = []
+        # Iterate through the nodes of self in topological order
+        # and form tuples containing sequences of gates
+        # on the same qubit(s).
+        nodes_seen = set()
+        for node in self.topological_op_nodes():
+            if len(node.qargs) == 1 and node.condition is None \
+                    and not node.cargs \
+                    and not node.op.is_parameterized() \
+                    and isinstance(node.op, Gate) \
+                    and node not in nodes_seen:
```

Comment:
One more performance question. Would it make sense to move `node not in nodes_seen` above the `is_parameterized` and `isinstance` checks? Because we walk the graph successors in the lower loop, the likelihood of hitting `node not in nodes_seen` for most circuits seems high, so we might be able to avoid a handful of unneeded `isinstance` checks.

Reply:
Sure I'll make this change. Heh, but fwiw this is a bit of a micro-optimization. I'm not sure it would be noticeable or not.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/5655/commits/90f9bce5b7ee930e1fbf88289f69a994ce6dfdf8